streaming/base/dataset.py

```diff
@@ -765,6 +787,154 @@ def _resample_streams(self, epoch: int) -> Tuple[NDArray[np.int64], NDArray[np.i
         sample_ids = np.concatenate(sample_ids).astype(np.int64)
         return shuffle_units, sample_ids
 
+    def _generate_work_stratified_batching(self, world: World, epoch: int,
```

Comment:
Same comments as in other PR:

Can we also move this method to partition folder since this basically partition the samples across 5D (num physical nodes, ranks per node, workers per rank, batches per worker, batch size) ? this way we keep specific functionality in their respective modules.

Reply:
yep, moving to `batching` folder